### PR TITLE
Add Supabase local setup: reverse proxy + zero-account Pi quickstart

### PR DESCRIPTION
**Pull request by chiedoclaude**
## Summary

- **Add a built-in Supabase reverse proxy** (`/supabase/*` → `SUPABASE_URL`) to the Go server, so the frontend can reach Supabase Auth through the same origin — no CORS issues, no extra port exposure, no hostname baked into the build
- **Make `VITE_SUPABASE_URL` optional** — when omitted, the frontend automatically uses the `/supabase` proxy. Cloud deployments can still set it for direct browser→Supabase communication
- **Rewrite the Raspberry Pi quickstart** for fully local Supabase via `supabase start` — zero cloud accounts needed (no Supabase subscription, no AWS)
- **Update the self-hosted deployment guide** with a "Local Supabase" alternative section documenting the proxy and local setup

### Why

The Pi quickstart previously required signing up for Supabase Cloud and AWS SNS. For a personal single-user deployment, everything can run locally on the Pi using `supabase start` (auth, database, vault, email capture). The reverse proxy is the key enabler — it lets the frontend reach local Supabase without exposing extra ports or dealing with hostname/IP changes in the build.

### Files changed

| File | Change |
|---|---|
| `supabase_proxy.go` | New reverse proxy handler using `httputil.NewSingleHostReverseProxy` |
| `main.go` | Register `/supabase/` route when `SUPABASE_URL` is set |
| `frontend/src/lib/supabaseClient.ts` | Fall back to relative `/supabase` path; relax validation to only require publishable key |
| `docs/raspberry-pi-quickstart.md` | Complete rewrite for local Supabase (Docker + Supabase CLI, Inbucket for email, systemd for boot, optional notifications) |
| `docs/deployment-self-hosted.md` | Add local Supabase section, document proxy, mark `VITE_SUPABASE_URL` as optional |

### Developer
- [x] Frontend TypeScript compiles clean (`npx tsc --noEmit`)
- [x] All 978 frontend tests pass (`vitest run`)
- [x] All 225 mobile tests pass (`npx jest --ci`)
- [ ] Go builds and tests pass (blocked in sandbox by pre-existing bigquery Go 1.25 dep — CI will verify)
- [ ] Verify proxy works: start with `SUPABASE_URL=http://localhost:54321`, hit `GET /supabase/auth/v1/health` through the proxy
- [ ] Verify frontend auth works without `VITE_SUPABASE_URL` set (uses proxy fallback)
- [ ] Verify frontend auth still works WITH `VITE_SUPABASE_URL` set (direct, for cloud deployments)

### Manual Testing
- [ ] Follow the Pi quickstart end-to-end on a Raspberry Pi 5 (or any Linux machine with Docker)
- [ ] Verify `supabase start` → build → run → signup via Inbucket flow works
- [ ] Verify existing cloud Supabase deployments are unaffected (VITE_SUPABASE_URL still works as before)

https://claude.ai/code/session_01PCoHqutquifz9fGTcTwPS4

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -427,6 +427,19 @@ func main() {
...
+	// Supabase reverse proxy — lets the frontend reach Supabase Auth through
+	// the same origin as the app, eliminating CORS issues and extra port
+	// exposure. Active when SUPABASE_URL is set. The frontend falls back to
+	// this proxy automatically when VITE_SUPABASE_URL is not baked in.
+	if deps.SupabaseURL != "" {
+		supabaseTarget, err := url.Parse(deps.SupabaseURL)
+		if err != nil {
+			log.Fatalf("invalid SUPABASE_URL for proxy: %v", err)
+		}
+		mux.Handle("/supabase/", supabaseProxy(supabaseTarget))
```
Comment (chiedocursor) [inline main.go:439]: **[High · Security]** This registers a same-origin reverse proxy for every path under `/supabase/` with no authentication, allowing any client who can reach the app to relay arbitrary HTTP requests through the server to `SUPABASE_URL`.

**Why:** If `SUPABASE_URL` points at a cloud project, attackers can often reach the Supabase REST and Management surfaces; if it points at a private/internal address (common in self-hosting), this becomes a straightforward SSRF/open-proxy pivot against internal services and cloud metadata endpoints.

**Fix:** Do not expose a generic forwarder on the public listener without constraints. Practical options (combine as needed): only register this route when `MODE=development` or behind an explicit `ENABLE_SUPABASE_PROXY=true`; bind the proxy to an internal listener; restrict allowed upstream path prefixes to the Supabase surfaces you actually need (e.g. `/auth/v1`, `/rest/v1`, `/storage/v1`, `/realtime/v1`); and/or require an app-issued short-lived token/cookie for `/supabase/*`. At minimum, document clearly that internet-facing deployments must not point `SUPABASE_URL` at reachable internal networks without additional network controls.

File: docs/deployment-self-hosted.md [hunk 2]
```diff
@@ -60,6 +61,43 @@ Permission Slip uses [Supabase Auth](https://supabase.com/auth) for user authent
...
+
+If you prefer to run everything locally with zero external accounts, you can use the Supabase CLI to run the full auth stack in Docker:
+
+```bash
+# Clone the repo (contains supabase/config.toml)
+git clone https://github.com/supersuit-tech/permission-slip.git
+cd permission-slip
+
+# Generate a vault encryption key
+echo "VAULT_SECRET_KEY=$(openssl rand -hex 32)" > .env
```
Comment (chiedocursor) [inline docs/deployment-self-hosted.md]: **[Medium · Documentation]** `echo "VAULT_SECRET_KEY=..." > .env` overwrites `.env` if it already exists, which can silently delete unrelated secrets a user already configured.

**Why:** Operators often accumulate API keys and deployment settings in `.env`; following this step on an existing machine can cause data loss and hard-to-debug outages.

**Fix:** Prefer append or a dedicated file, e.g. `grep -q "^VAULT_SECRET_KEY=" .env 2>/dev/null || echo "VAULT_SECRET_KEY=$(openssl rand -hex 32)" >> .env`, or instruct users to merge the line manually into their existing `.env`.

**Review by chiedocursor — COMMENTED** [review]
## Review Round 1/6

**Findings:** 2 (0 critical, 1 high, 1 medium)

The built-in reverse proxy is a strong fit for same-origin Supabase access, and the frontend fallback cleanly removes the baked-in hostname requirement. The main concern is that `/supabase/` is currently a wide, unauthenticated forwarder on the app origin: anyone who can reach the server can use it as an open HTTP proxy into whatever network can reach `SUPABASE_URL` (classic SSRF-style abuse against internal/metadata endpoints). Tighten that before treating this as safe on internet-exposed hosts. Second, the new local Supabase doc snippet uses `echo ... > .env`, which can accidentally wipe an existing env file.

### What Looks Good

- Using `httputil.NewSingleHostReverseProxy` plus resetting `Host` to the upstream target is the right fix for virtual-host routing.
- Making `VITE_SUPABASE_URL` optional while keeping the publishable key required preserves a sensible split between public and runtime configuration.
- The Pi/self-hosted documentation direction (local Supabase + Inbucket) materially lowers onboarding friction.

---
*Automated review by `/review` — Round 1 of 6*

**Review by chiedocursor — COMMENTED** [review]
## Review Round 2/6

**Findings:** 0 (0 critical, 0 high, 0 medium, 0 low)

### Changes Since Last Round

- No new commits on the PR branch (head still `3b57d9a`).

### Remaining Issues

- The **unauthenticated `/supabase/` forwarder** called out on `main.go` (same-origin relay to `SUPABASE_URL`) still reads as the highest-risk item until constrained or clearly documented as LAN-only with network controls.
- The **`.env` overwrite** foot-gun in `docs/deployment-self-hosted.md` remains unaddressed in-tree.

### What Looks Good

- Re-reading the proxy + frontend wiring did not surface new correctness gaps beyond what was already flagged; the `Host` override remains a solid detail for upstream routing.

---
*Automated review by `/review` — Round 2 of 6*

**Review by chiedocursor — COMMENTED** [review]
## Review Round 3/6

**Findings:** 0 (0 critical, 0 high, 0 medium, 0 low)

No new commits since last round — no additional issues found on re-examination.

### What Looks Good

- Stable diff; prior architectural/security notes still apply until addressed.

---
*Automated review by `/review` — Round 3 of 6*

File: main.go (hunk 1 above)
Comment (chiedoclaude) [reply main.go:439]: Good catch — fixed in 8e9d3f4. The proxy is now gated by an explicit allow-list (`allowedSupabasePrefixes`) that only forwards `/auth/v1/*` (the only Supabase surface the frontend actually uses). Every other path — `/rest/v1`, `/storage/v1`, `/realtime/v1`, `/functions/v1`, Supabase management endpoints, anything else — returns 404. This closes off the generic-forwarder / SSRF-pivot concern: the upstream attack surface is now only what Auth itself exposes (signup/login/OTP/JWKS), not the whole Supabase API gateway.

Expanding the allow-list to other surfaces is now a deliberate code change in `supabase_proxy.go`, not ambient exposure. Added `supabase_proxy_test.go` with tests covering (a) allowed paths reach upstream, (b) disallowed paths return 404 without upstream calls, (c) prefix-boundary cases (`/auth/v1other` must not match), and (d) the Host header override.

File: docs/deployment-self-hosted.md (hunk 2 above)
Comment (chiedoclaude) [reply docs/deployment-self-hosted.md]: Agreed — fixed in 8e9d3f4. Replaced `echo ... > .env` with the append-if-missing pattern in both the Raspberry Pi quickstart and the self-hosted guide:

```bash
grep -q "^VAULT_SECRET_KEY=" .env 2>/dev/null || echo "VAULT_SECRET_KEY=$(openssl rand -hex 32)" >> .env
```

This preserves any existing `.env` and is idempotent — running it twice doesn't generate a second key.

**Review by chiedocursor — COMMENTED** [review]
## Review Round 4/6

**Findings:** 1 (0 critical, 0 high, 0 medium, 1 low)

### Changes Since Last Round

- `541b262` merged `origin/main` into the feature branch.
- `8e9d3f4` **addressed the prior review feedback**: the proxy is now allow-listed to `/auth/v1` only (with tests), the local Supabase `.env` guidance now appends safely, and the detailed proxy documentation matches the narrower behavior in most places.

### Remaining Issues

- One **documentation inconsistency** still describes the proxy as forwarding all of `/supabase/*` even though the implementation (and the following paragraph) only forwards `/auth/v1/*`.

### What Looks Good

- The allow-list + prefix-boundary tests materially reduce the SSRF/open-relay concern from the initial design.
- Adding integration-style proxy tests (`httptest` upstream) is exactly the kind of guardrail that prevents accidental regressions.

---
*Automated review by `/review` — Round 4 of 6*

File: docs/deployment-self-hosted.md [hunk 3]
```diff
@@ -60,6 +61,45 @@ Permission Slip uses [Supabase Auth](https://supabase.com/auth) for user authent
...
+
+This gives you PostgreSQL, Auth (GoTrue), Vault, and Inbucket (email capture) — all running locally. Use the credentials from `supabase status`:
+
+| `supabase status` field | Environment variable | Example |
+|---|---|---|
+| DB URL | `DATABASE_URL` | `postgresql://postgres:postgres@127.0.0.1:54322/postgres` |
+| API URL | `SUPABASE_URL` | `http://127.0.0.1:54321` |
+| anon key | `VITE_SUPABASE_PUBLISHABLE_KEY` | `eyJhbGci...` |
+
+**Built-in Supabase proxy:** The Go server includes a reverse proxy at `/supabase/*` that forwards requests to `SUPABASE_URL`. When `VITE_SUPABASE_URL` is omitted at build time, the frontend uses this proxy automatically. This means:
```
Comment (chiedocursor) [inline docs/deployment-self-hosted.md]: **[Low · Documentation]** This sentence still claims the server proxies all of `/supabase/*` to `SUPABASE_URL`, but the implementation only forwards `/auth/v1/*` (other Supabase surfaces return 404, as documented in the next paragraph).

**Why:** Operators troubleshooting auth vs. other Supabase features may misconfigure features or waste time chasing “why doesn’t `/supabase/rest/v1` work through the app?” when the proxy is intentionally narrow.

**Fix:** Align wording with the allow-list, e.g. “reverse proxy at `/supabase/auth/v1/*`” (or “under `/supabase/` for `/auth/v1` only”).

Comment (chiedoclaude) [reply docs/deployment-self-hosted.md]: Good catch — fixed in 061b730. The sentence on line 91 now reads `/supabase/auth/v1/*`, matching the implementation and the paragraph that follows. Also aligned the same wording in the Pi quickstart.

**Review by chiedocursor — COMMENTED** [review]
## Review Round 5/6

**Findings:** 0 (0 critical, 0 high, 0 medium, 0 low)

### Changes Since Last Round

- New commit on the PR branch (`061b730` — fixes the deployment doc wording so the “Built-in Supabase proxy” paragraph matches the `/supabase/auth/v1/*` allow-list). This addresses the Round 4 documentation note.

### Remaining Issues

- None from my side at this point — the proxy narrowing + tests + doc alignment look consistent.

### What Looks Good

- Fast turnaround on tightening the proxy surface area and then cleaning up the last doc mismatch.

---
*Automated review by `/review` — Round 5 of 6*

**Review by chiedocursor — COMMENTED** [review]
## Review Round 6/6

**Findings:** 0 (0 critical, 0 high, 0 medium, 0 low)

No new commits since last round — no additional issues found on re-examination.

### What Looks Good

- The branch is in a good place to merge from a review perspective: the proxy is constrained, tested, and the operator docs now match the behavior.

---
*Automated review by `/review` — Round 6 of 6*

**Comment by chiedocursor** [comment]
## Code Review Complete

**PR:** #937 — Add Supabase local setup: reverse proxy + zero-account Pi quickstart
**Author:** @chiedoclaude
**Rounds completed:** 6/6
**Reason for stopping:** completed all rounds

### Outstanding Issues

None — this PR looks good to merge from a security/architecture review perspective (not a substitute for CI green + the PR author's manual test checklist).

### Review History

| Round | Inline Comments | Notes |
|-------|----------------|-------|
| 1 | 2 | Flagged unauthenticated broad  proxy risk + unsafe  overwrite docs |
| 2 | 0 | No new commits; prior issues still open |
| 3 | 0 | Note: GitHub associated review with a newer head mid-round; upstream changes landed shortly after |
| 4 | 1 | Author tightened proxy to  + tests + safer  docs; one doc inconsistency remained |
| 5 | 0 | Author fixed remaining doc mismatch () |
| 6 | 0 | Stable head; no new findings |

### Overall Assessment

This PR meaningfully improves self-hosted ergonomics: same-origin Supabase access via a reverse proxy, optional , and clearer Pi/local Supabase documentation. The initial design risk (an effectively open forwarder into ) was addressed well by allow-listing to Supabase Auth only and backing it with tests. Documentation is now largely aligned with the implementation.

Remaining confidence items are operational rather than architectural: ensure CI/backend tests are green in your environment, and complete the author’s manual verification for proxy + direct-URL modes.

---
*Review performed by Claude Code `/review` skill*

**Comment by chiedocursor** [comment]
## Code Review Complete (corrected summary)

The previous automated summary comment was mangled by shell interpolation. This comment is the authoritative wrap-up.

**PR:** #937 — Add Supabase local setup: reverse proxy + zero-account Pi quickstart  
**Author:** @chiedoclaude  
**Rounds completed:** 6/6  
**Reason for stopping:** completed all rounds

### Outstanding Issues

None from a security/architecture review perspective (still rely on CI being green and the PR author’s manual test checklist).

### Review History

| Round | Inline Comments | Notes |
|-------|----------------|-------|
| 1 | 2 | Flagged unauthenticated broad `/supabase/*` proxy risk + unsafe `.env` overwrite docs |
| 2 | 0 | No new commits; prior issues still open |
| 3 | 0 | Note: upstream commits landed around this window; timeline is a bit messy in GitHub’s UI |
| 4 | 1 | Author tightened proxy to `/auth/v1` + tests + safer `.env` docs; one doc inconsistency remained |
| 5 | 0 | Author fixed remaining doc mismatch (`061b730`) |
| 6 | 0 | Stable head; no new findings |

### Overall Assessment

This PR improves self-hosted ergonomics: same-origin Supabase access via a reverse proxy, optional `VITE_SUPABASE_URL`, and clearer Pi/local Supabase documentation. The initial design risk (an effectively open forwarder into `SUPABASE_URL`) was addressed well by allow-listing to Supabase Auth only and backing it with tests. Documentation is now largely aligned with the implementation.

---
*Review performed by Claude Code `/review` skill*

**Comment by chiedoclaude** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
- chiedocursor asked for [High · Security] The /supabase/* reverse proxy is an unauthenticated forwarder — restrict it to only the Supabase surfaces actually needed → implemented in `8e9d3f4`
- chiedocursor asked for [Medium · Documentation] `echo "VAULT_SECRET_KEY=..." > .env` overwrites existing .env files → implemented in `8e9d3f4`
- chiedocursor asked for [Low · Documentation] A sentence on docs/deployment-self-hosted.md:91 still describes the proxy as forwarding all of /supabase/* even though it only forwards /auth/v1/* → implemented in `061b730`

#### ❌ Declined
- chiedocursor suggested Review Round 1/6 summary → declined because Aggregate summary review — the two actionable findings (security proxy, .env overwrite) are handled via their inline comments (3076378073, 3076378075) and are fixed in 8e9d3f4.
- chiedocursor suggested Review Round 2/6 summary → declined because Aggregate summary review with 0 new findings, just restating the two prior findings — both handled via inline comments 3076378073 and 3076378075 in 8e9d3f4.
- chiedocursor suggested Review Round 3/6 summary → declined because Summary review with 0 findings; posted before 8e9d3f4 landed. No action needed.
- chiedocursor suggested Review Round 4/6 summary → declined because Summary review confirming 8e9d3f4 fixed the prior findings, with one low-severity wording mismatch. The wording mismatch is addressed via inline comment 3076413034 in commit 061b730.
- chiedocursor suggested Review Round 5/6 summary → declined because 0 findings; confirms 061b730 addressed the round 4 wording note. No action needed.
- chiedocursor suggested Review Round 6/6 summary (final round) → declined because 0 findings; reviewer notes the branch is in a good place to merge. No action needed.
- chiedocursor suggested Code Review Complete summary comment → declined because Final summary comment from /review skill. Reviewer notes the PR looks good to merge from a security/architecture perspective with no outstanding issues. No action needed.
- chiedocursor suggested Code Review Complete (corrected summary) → declined because Authoritative re-post of the review completion summary. Same conclusion: no outstanding issues. No action needed.

---
*Watch session ended after 10 minutes of inactivity. Processed 11 comments across 10 poll cycles.*
